core/ledger/ledgermgmt/snapshot_service.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package ledgermgmt
+
+import (
+	"context"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/golang/protobuf/ptypes/empty"
+	pb "github.com/hyperledger/fabric-protos-go/peer"
+	"github.com/hyperledger/fabric/core/ledger"
+	"github.com/pkg/errors"
+)
+
+// Snapshot Service implements SnapshotServer grpc interface
+type SnapshotService struct {
+	LedgerGetter LedgerGetter
+}
+
+// LedgerGetter gets the PeerLedger associated with a channel.
+type LedgerGetter interface {
+	GetLedger(cid string) ledger.PeerLedger
+}
```

Comment:
This looks odd to me that this service impl is in ledgermgmt package in which LegderMgr keeps track of all the ledgers and a dependency is being introduced for getting access to a ledger. This round-about dependency where the single package is create and opens ledgers and at the same time needs the access to the same object via an external dependency may cause confusions. I would prefer either
- Have this code in a separate package (snapshotgrpc?) and then have it as is
- If in this package, then better to implement these functions directly on LedgerMgr

I tend to inclined towards the former as this function is not directly related to ledgermgmt.